mlflow/server/handlers.py

```diff
@@ -1363,6 +1363,23 @@ def _delete_registered_model_tag():
     return _wrap_response(DeleteRegisteredModelTag.Response())
 
 
+def _validate_source(source: str, run_id: str) -> None:
+    if not is_local_uri(source):
+        return
+
+    if run_id:
+        store = _get_tracking_store()
+        run = store.get_run(run_id)
+        if source.startswith(run.info.artifact_uri):
```

Comment:
This line needs to be fixed. `/a/b/c/d` (source) should be considered within `file:////a/b/c` (run.info.artifact_uri)